### PR TITLE
chore(deps): upgrade webpack 4 style loaders to webpack 5

### PR DESCRIPTION
**Pull request by stephenliang**
The Code.org code base uses some webpack 4 style loaders which were deprecated and merged into the [webpack 5 asset modules feature](https://webpack.js.org/guides/asset-modules/). These loaders are no longer maintained and have some critical dependency issues. To mitigate these issues, migrate the loaders to the webpack 5 feature.

**What was removed?**

* url-loader: Used to base64 encode an image. This is now automatically done via `asset/inline` and therefore the dependency can be removed after swapping to `asset/inline`.
* file-loader: This is not used in the code base except in unit tests. Jest handles this automatically, and therefore the dependency can be removed.
* raw-loader: This is not used in the code base, and therefore the dependency can be removed.

This addresses a majority of the issues in [dependabot 325](https://github.com/code-dot-org/code-dot-org/security/dependabot/325) but doesn't resolve it completely.

## Testing story

There is only one usage of `url-loader` in the code base, the app lab turtle. To test this, I inspected the production base64 encoding of the turtle and compared it with localhost. Both turtles produced the same output.

## Deployment strategy

Merge this at night on a standalone DTT to ensure no eyes mismatches as this updates image generation in bulk

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/applab/applabTurtle.js
Comment (wilkie) [inline apps/src/applab/applabTurtle.js:3]: Wow... it is only used in one place? Good find.

File: apps/webpack.config.js
Comment (wilkie) [inline apps/webpack.config.js]: This comment is unnecessary now I believe

**Review by wilkie — COMMENTED** [review]
I'm a little intrigued and probably a little ignorant. You only found one use of `url-loader` yet this rule seems to apply to a large number of images. What will be the broad difference in behavior for those images? Are we concerned about, as this section was concerned about, the behavior for large files?

**Comment by stephenliang** [comment]
> I'm a little intrigued and probably a little ignorant. You only found one use of `url-loader` yet this rule seems to apply to a large number of images. What will be the broad difference in behavior for those images? Are we concerned about, as this section was concerned about, the behavior for large files?

There is no fundamental difference, in webpack v4, `url-loader` was applied to the images which converted the following regex to base64 encoded inline images:

```
        test: /\.(png|jpg|jpeg|gif|svg)$/,
        include: [
          p('static'),
          p('src'),
          p('test'),
          p('../dashboard/app/assets/images'),
        ],
```

In webpack v5, the equivalent is provided in [webpack asset modules](https://webpack.js.org/guides/asset-modules/) and is documented:


> asset/inline exports a data URI of the asset. Previously achievable by using url-loader.

The images should have the same base64 encoding prior to this change and after this change, and I'll probably merge this at night on a standalone DTT to ensure no eyes mismatches.

**Review by wilkie — APPROVED** [review]
LGTM! 👍

**Comment by stephenliang** [comment]
I'm targeting to merge this July 17th after the DTP and will monitor the DTT for baseline changes.

**Comment by stephenliang** [comment]
[No baseline changes were detected when doing the eyes test](https://codedotorg.slack.com/archives/C03CM903Y/p1721257205606979) - going to merge this in now!

**Pull request merged**
